### DISCUSSION TITLE
Test VS prod

### DISCUSSION THREAD

**Discussion by ScottJOster** [Netflix/conductor#3334] [Q&A]
Hi,

I was wondering how to best separate workflows in terms of test and production.

Eg if I'm creating and firing workflows on test environment how then do I transfer workflows to prod- given that the tasks such as http have been configured to talk to test microservices etc.

Some comments on here suggest live prod traffic and dev traffic are used in a shared env?

In a nutshell - basic guidance on separating dev/test and prod with conductor.

Any help from Netflix / the community appreciated !

Thanks

**Comment by dcore94** [comment]
In our case we use different Conductor Docker stacks for dev, staging and production infrastructures.
We upload workflows with automated procedures based on Ansible and we use Ansible's templating facility to "patch" an initial SET_VARIABLE task where we set infrastructure specific information such as service endpoints.

**Reply by ScottJOster** [reply]
Thanks a lot dcore I'll consider this ! Be interesting to see how others handle also! 

Thanks again 😎

**Comment by boney9** [comment]
If you'd like to have a single stack for both prod and dev, you could leverage domains

for ex: for all your prod workflows start the workflow with the task domain set as `prod`
And for other environments you can use a different label = `test` or `staging` for example.

For the workers listening to `prod` make sure you are listening with the same label as you configured and similarly for the test.

**Reply by ScottJOster** [reply]
Thanks Boney much appreciated!
